### ISSUE TITLE
save as PDF

### ISSUE THREAD

**Issue by softboy99** [EvotecIT/OfficeIMO#36]

**Comment by PrzemyslawKlys** [comment]
Short answer: 
Not really planned. 

Long answer: 
It's just not trivial. PDF has different structure so one effectively writes it from scratch. There are other projects that allow creating PDFS from C#

**Comment by softboy99** [comment]
yes, There are other projects that allow creating PDFS, but there're no project for docx2pdf

**Comment by PrzemyslawKlys** [comment]
Because it's not trivial. You need to read DOCX and translate all that into PDF. Two separate approaches, two different code bases. I've some idea how to resolve it, but probably it won't be in the OfficeIMO project, but completely new one. Don't have your hopes up.

**Comment by softboy99** [comment]
Hi,
you can create an new project and share your idea, we can do that together and invite other contributors

**Comment by softboy99** [comment]
> @softboy99 Wouldn't that library do, what you need? https://github.com/smartinmedia/Net-Core-DocX-HTML-To-PDF-Converter

Hi,
we need to use it in server environment,your lib depends on libreoffice.exe.

**Comment by softboy99** [comment]
In server environment, how many concurrent operations can be holded? Each client request creat an libre office process instance? Your way has nothing difference with office client automation.

**Comment by softboy99** [comment]
> Because it's not trivial. You need to read DOCX and translate all that into PDF. Two separate approaches, two different code bases. I've some idea how to resolve it, but probably it won't be in the OfficeIMO project, but completely new one. Don't have your hopes up.

Hi,
@PrzemyslawKlys ,Could you please open up an project and share your "some idea"? So that we can attracts others to contribute

**Comment by tmheath** [comment]
Popping in to mention.... I'm not doing it, if I were though I'd hash something dirty together using Pandoc.
Last I knew it didn't support all features of word though if that is a concern.
https://pandoc.org/MANUAL.html#option--from

**Comment by TopperDEL** [comment]
Just in case anyone needs that info: I'm using Gotenberg in a docker container to convert from DOCX to PDF. In my case that was a viable solution.

**Comment by PrzemyslawKlys** [comment]
Have you tried new SaveAsPdf option?

**Comment by TopperDEL** [comment]
Not yet. Didn't see it. But I need Gotenberg for the conversion of other dataypes (images, xls and others), too. But thanks for the hint!

**Comment by PrzemyslawKlys** [comment]
I added basic SaveAsPdf, SaveAsHtml, SaveAsMarkdown, and load markdown and load html. It's not currently released but please feel free to provide feedback.
